go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -456,7 +458,14 @@ func (oc *SecondaryLayer2NetworkController) addUpdateLocalNodeEvent(node *corev1
 					errs = append(errs, err)
 					oc.gatewaysFailed.Store(node.Name, true)
 				} else {
-					oc.gatewaysFailed.Delete(node.Name)
+					if util.IsNetworkSegmentationSupportEnabled() && oc.IsPrimaryNetwork() {
```

Comment:
what about deleteNodeEvent ? do we need to delete local pod subnet SNATs explicitly there ?

Reply:
a node delete I thought about it.. in IC will totally delete the database and switches and routers and everything.. so ovnkube-node is totally gone.. and given we don't support L2 on nonIC we should be good.